### ISSUE TITLE
Rule Request: check for method existence in #selector

### ISSUE THREAD

**Issue by sammy-SC** [realm/SwiftLint#2084]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Rule Request

If this is a new rule request, please ignore all sections below this one, format
this issue's title as `Rule Request: [Rule Name]` and describe:

Describe: 
#selector() doesn't warn you if you reference method from different class. In class A you could be adding target to button `button.addTarget(self, action: #selector(B.didTap), for: .touchUpInside)` and as long as `B.didTap` is accessible this code compiles but on tap of the button crashes the app. I would keep this rule simple and it would only check for existence of method in case target is being added to `self`.

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

I don't have link to an existing discussion.

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

What would trigger: 
```
final class A: UIViewController {
  override func viewDidLoad() {
    super.viewDidLoad()
    let button = UIButton()
    button.addTarget(self,
                                   action: #selector(B.didTap),
                                   for: .touchUpInside)
  }
}
```

what wouldn't trigger:
```
final class A: UIViewController {
  override func viewDidLoad() {
    super.viewDidLoad()
    let button = UIButton()
    button.addTarget(self,
                                   action: #selector(A.didTap),
                                   for: .touchUpInside)
  }
}
```

3. Should the rule be configurable, if so what parameters should be configurable?

This rule should not be configurable.

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a
   rule as opt-in.

I believe this rule should be enabled by default.

**Comment by marcelofabri** [comment]
Should this only work with `UIControl` APIs? What about other classes that use target-action (e.g. gesture recognizers)?

**Comment by sammy-SC** [comment]
@marcelofabri you are right, it should work with any class that uses target-action.

However I realised one problem with implementation, what if the method in action is in subclass? That way the method is invisible to SwiftLint but won't crash the application. Is there a way to work around this?

**Comment by marcelofabri** [comment]
@sammy-SC I don’t think that’s a major use case. The rule should be opt-in anyways, so I’m fine with some false positives.
